sourced/ml/core/tests/test_token_parser.py

```diff
@@ -253,6 +253,57 @@ def test_pickle(self):
         self.assertEqual(tp.stem("embedding"), "embed")
 
 
+class NNTokenParserTests(unittest.TestCase):
+    def setUp(self):
+        self.tp = TokenParser(stem_threshold=4, max_token_length=20,
+                              attach_upper=False, use_nn=True)
+        self.tp._single_shot = False
+
+    def test_process_token(self):
+        self.tp.max_token_length = 100
+
+        tokens = [
+            ("ONLYCAPS", ["only", "caps"]),
+            ("nocaps", ["no", "caps"]),
+            ("UpperCamelCase", ["upper", "camel", "case"]),
+            ("camelCase", ["camel", "case"]),
+            ("FRAPScase", ["frap", "case"]),
+            ("SQLThing", ['sql', 'thing']),
+            ("_Astra", ["astra"]),
+            ("CAPS_CONST", ["caps", "const"]),
+            ("_something_SILLY_", ["someth", "silli"]),
+            ("blink182", ["blink"]),
+            ("FooBar100500Bingo", ["foo", "bar", "bingo"]),
+            ("Man45var", ["man", "var"]),
+            ("method_name", ["method", "name"]),
+            ("Method_Name", ["method", "name"]),
+            ("101dalms", ["dalm"]),
+            ("101_dalms", ["dalm"]),
+            ("101_DalmsBug", ['dalmsbug']),
+            ("101_Dalms45Bug7", ['dalmsbug']),
+            ("wdSize", ['wd', 'size']),
+            ("Glint", ["glint"]),
+            ("foo_BAR", ['foobar']),
```

Comment:
The logic should be:

1. Split by non-alpha chars (numbers, underscore, etc.)
2. Split each part by NN

So this test is invalid, must be "foo" and "bar"

Reply:
I agree, I think this should rather be fixed inside the NN splitter class itself wdyt ?